### DISCUSSION TITLE
[AssetMapper] importmap and earlyhints?

### DISCUSSION THREAD

**Discussion by Nayte91** [symfony/symfony#53220] [Q&A] [answered]
Hello,
I'm a bit limited in my possibilites to try things out, but I was wondering something (possibly) interesting:

Browsers are now able to parse an importmap, a JSON with a list of assets that should be loaded in order to display the page correctly. AssetMapper uses it.

Apparently, importmap MUST be inlined in the DOM's <head>, in a <script> tag with type="importmap".

103 earlyhints is a new Response's status that can be sent to help browser to load assets for and BEFORE the incoming DOM.

What if we send an importmap in HTTP header link? Or in 103 earlyHints? Because of specifications, it SHOULD not work?

It seems that we can't serialize the JSOn in earlyhint head, so it's more like sending a link to importmap.json as: importmap.

If (importmap can be early-hinted) {
   then it's a deluxe choice to send a bunch of asset resources and dependencies,
} else {
   the 2 features just fight each other by limiting their scope and usability mutually.
}

**Do I miss something**? Or should importmap be early-hintable? Are they actually? If not, how can we suggest this change?

**Comment by MatTheCat** [comment, marked as answer]
Hello,

I’m not aware of importmaps not being early-hintable, but as you can read at https://github.com/WICG/import-maps#installation:

> The inline form of import maps is strongly recommended for best performance. This is similar to the best practice of [inlining critical CSS](https://developers.google.com/speed/docs/insights/OptimizeCSSDelivery); both types of resources block your application from doing important work until they're processed, so introducing a second network round-trip (or even disk-cache round trip) is a bad idea.
